### PR TITLE
[TASK] Streamline database query builder chapter

### PR DESCRIPTION
**Pull request by brotkrueml**
Releases: main, 11.5

### CODE REVIEW

File: Documentation/ApiOverview/Database/QueryBuilder/Index.rst [hunk 1]
```diff
@@ -1,1133 +1,1234 @@
...
+    injections!
 
-   From security point of view, the documentation of
-   :ref:`->createNamedParameter() <database-query-builder-create-named-parameter>` and
-   :ref:`->quoteIdentifier() <database-query-builder-quote-identifier>` are an absolute **must read and follow** section.
-   Make very sure this is understood and use this for **each and every query** to prevent SQL injections!
 
+The query builder comes with a list of small methods:
 
-The `QueryBuilder` comes with a happy little list of small methods:
```
Comment (linawolf) [inline Documentation/ApiOverview/Database/QueryBuilder/Index.rst:25]: Lets not remove the "happy little list" this is O-tone lolli and I love it!

File: Documentation/ApiOverview/Database/QueryBuilder/Index.rst [hunk 2]
```diff
@@ -1,1133 +1,1234 @@
...
-   // DELETE FROM `tt_content` WHERE `bodytext` = 'klaus'
-   $queryBuilder = GeneralUtility::makeInstance(ConnectionPool::class)->getQueryBuilderForTable('tt_content');
-   $affectedRows = $queryBuilder
-      ->delete('tt_content')
-      ->where(
-         $queryBuilder->expr()->eq('bodytext', $queryBuilder->createNamedParameter('klaus'))
-      )
-      ->executeStatement();
+    // DELETE FROM `tt_content` WHERE `bodytext` = 'klaus'
+    $queryBuilder = $this->connectionPool->getQueryBuilderForTable('tt_content');
```
Comment (linawolf) [inline Documentation/ApiOverview/Database/QueryBuilder/Index.rst:244]: Please include injection of the ConnectionPool. newbies dont't always have that context

Comment (linawolf) [reply Documentation/ApiOverview/Database/QueryBuilder/Index.rst:244]: yeah I can see how we would need them in many places. However if we just assume `$this->connectionPool` we don't even tell the type

Comment (brotkrueml) [reply Documentation/ApiOverview/Database/QueryBuilder/Index.rst:244]: Yes, using full classes would blow up examples. First I haven't `$this->connectionPool->getQueryBuilderForTable()` in the snippets, but is relevant as the query builder instance depends on the table name to get the correct connection.

In the query builder chapter at least the instantiation section should be read. There are so many cross-references between the sections, so you should read at least the introduction to get a clue what's going on.

Anyway, we can add a comment in each example (just before `$this->connectionPool->...` to look up the instatiation section on top of the chapter.

Comment (linawolf) [reply Documentation/ApiOverview/Database/QueryBuilder/Index.rst:244]: I would prefere to have a comment linking how to get the connection Pool, this is a topic where many newbies might get frustrated on.

Comment (brotkrueml) [reply Documentation/ApiOverview/Database/QueryBuilder/Index.rst:244]: Yup, but a newbie should read the chapter and not just copy/paste. As I said, many cross-references between the different sections.

But I will add a comment in the snippets.

Comment (linawolf) [reply Documentation/ApiOverview/Database/QueryBuilder/Index.rst:244]: I would add them outside the snippets and link them properly?

We have the problem that ppl are searching for something and then they will land on one of these pages. We cannot assume they start with A.

Comment (brotkrueml) [reply Documentation/ApiOverview/Database/QueryBuilder/Index.rst:244]: Okay, then above all snippets links... That'll be fun ...

query builder is special. ppl should know what they are doing.

Comment (brotkrueml) [reply Documentation/ApiOverview/Database/QueryBuilder/Index.rst:249]: Done it below, above does not fit.

**Review by linawolf — APPROVED** [review]

**Pull request merged**